w3af/plugins/auth/detailed.py

```diff
@@ -58,7 +58,18 @@ def login(self):
 
         try:
             functor = getattr(self._uri_opener, self.method)
-            functor(self.auth_url, data)
+
+            while True:
+                # actually file the auth HTTP reuqest
+                r = functor(self.auth_url, data)
+                # follow redirects if the feature is enabled
+                if self.follow_redirects and r.get_code() % 300 < 10:
+                    self.auth_url.set_path(r.get_headers().iget('location')[0])
```

Comment:
There are problems with this approach:
 * Changing the auth_url, which is a user configured setting might bring issues in the future
 * Only "Location" redirects are supported, when there are many other types

Redirecting is difficult, there is a handler for it in w3af (which inherits from the urllib2 handler) https://github.com/andresriancho/w3af/blob/master/w3af/core/data/url/handlers/redirect.py

This code also follows redirects outside the initial application URL, which is usually something we don't want, but I understand where you're coming from with this.

Finally, the next "GET" to the redirect target is only done after some considerable time happens. Thus the initial call to the login() method will never work.

If you've got some minutes I would love to talk to you about this [on IRC](http://w3af.org/community) so we can find a way to make this feature available to all with the best practices